File: DraftRightMobile/android/app/src/main/kotlin/com/draftright/keyboard/Composer.kt
package com.draftright.keyboard

sealed class ComposeResult {
    object PassThrough : ComposeResult()
    data class Commit(val text: String) : ComposeResult()
    data class Composing(val text: String) : ComposeResult()
    object Consumed : ComposeResult()
}

interface Composer {
    fun onKey(char: Char): ComposeResult
    fun onBackspace(): ComposeResult
    fun reset()
    fun currentComposingText(): String
}

File: DraftRightMobile/ios/DraftRightKeyboardCore/Sources/DraftRightKeyboardCore/Composer.swift
import Foundation

public enum ComposeResult: Equatable {
    case passThrough
    case commit(String)
    case composing(String)
    case consumed
}

public protocol Composer: AnyObject {
    func onKey(_ char: Character) -> ComposeResult
    func onBackspace() -> ComposeResult
    func reset()
    func currentComposingText() -> String
}
